packages/nimble-components/src/utilities/models/converter.ts

```diff
@@ -0,0 +1,17 @@
+import { booleanConverter, ValueConverter } from '@microsoft/fast-element';
+
+export const optionalBooleanConverter: ValueConverter = {
+    toView(value: unknown): string {
+        return typeof value === 'boolean' ? value.toString() : '';
+    },
+
+    fromView(value: string): boolean | null | undefined {
+        if (value === undefined) {
+            return undefined;
+        }
+        if (value === null) {
+            return null;
+        }
+        return booleanConverter.fromView(value) as boolean;
```

Comment:
Seems weird to not be symmetric if we are leveraging fast converters.

And the [fast converter](https://github.com/microsoft/fast/blob/c2b4076a8d1d527551b3c6680c17ba6fd030c147/packages/web-components/fast-element/src/components/attributes.ts#L77) is doing weird cases we don't care about like the value number `0` and falling back to true for misconfigurations. true seems like the wrong fallback behavior over absent.

FAST's reasoning is that any value for an attribute should be true `<my-el val>`, `<my-el val="">`, `<my-el val="boo!">`, `<my-el val="false">`. They are putting in exceptions to be friendly I guess where they are going out of spec to treat `<my-el val="false">` as a special case for false. Irregardless we have already stated normal boolean behavior is not what we expect so relying on this is wrong.

Lets handle our expected cases with a more appropriate fallback. `'true'`, `'false'`, `true`, `false` and fallback to `undefined` instead of `true`.

Reply:
Unresolving as the fromView does not handle the cases give, see:

> Lets handle our expected cases with a more appropriate fallback. 'true', 'false', true, false and fallback to undefined instead of true.

missing the boolean `true` and boolean `false` cases.

Reply:
The requirement that fromView handles the possible property values (the actual booleans) comes from my understanding of this topic: https://www.fast.design/docs/fast-element/defining-elements#customizing-attributes

Specifically:

> When the property value changes, the converter's fromView method will also be called, ensuring that the type is correct. After this, the mode will be determined. If the mode is set to reflect then the converter's toView method will be called to allow the type to be formatted before writing to the attribute using setAttribute.

Which seems to indicate fromView needs to handle property values and attribute string values.